taca/analysis/analysis.py

```diff
@@ -43,28 +44,34 @@ def get_runObj(run):
         logger.warn("Problems parsing the runParameters.xml file at {}. "
                     "This is quite unexpected. please archive the run {} manually".format(rppath, run))
     else:
-        # This information about the run type (with HiSeq2.5 applicationaName does not work anymore,
-        # but as for a long time we will have instruments not updated I need to find out something that works
-        try:
-            # Works for recent control software
-            runtype = rp.data['RunParameters']["Setup"]["Flowcell"]
-        except KeyError:
-            # Use this as second resource but print a warning in the logs
-            logger.warn("Parsing runParameters to fecth instrument type, "
-                        "not found Flowcell information in it. Using ApplicationName")
-            # here makes sense to use get with default value "" ->
-            # so that it doesn't raise an exception in the next lines
-            # (in case ApplicationName is not found, get returns None)
-            runtype = rp.data['RunParameters']["Setup"].get("ApplicationName", "")
-
-        if "HiSeq X" in runtype:
+        #do a case by case test becasue there are so many version of RunParameters that there is no real other way
+        if "Setup" in rp.data['RunParameters']:
+            #this is the HiSeq2500, MiSeq, and HiSeqX case
+            try:
+                # Works for recent control software
+                runtype = rp.data['RunParameters']["Setup"]["Flowcell"]
+            except KeyError:
+                # Use this as second resource but print a warning in the logs
+                logger.warn("Parsing runParameters to fecth instrument type, "
+                            "not found Flowcell information in it. Using ApplicationName")
+                # here makes sense to use get with default value "" ->
+                # so that it doesn't raise an exception in the next lines
+                # (in case ApplicationName is not found, get returns None)
+                runtype = rp.data['RunParameters']["Setup"].get("ApplicationName", "")
+        elif "Application" in rp.data['RunParameters']:
+            runtype = rp.data['RunParameters'].get("Application", "")
+
```

Comment:
Could we make the above `if/else` loop to 
```
runtype = rp.data['RunParameters'].get("Application", "")
if "Setup" in rp.data['RunParameters']:
    #this is the HiSeq2500, MiSeq, and HiSeqX case
    try:
       # Works for recent control software
       runtype = rp.data['RunParameters']["Setup"]["Flowcell"]
    except KeyError:
        # Use this as second resource but print a warning in the logs
        logger.warn("Parsing runParameters to fecth instrument type, "
                             "not found Flowcell information in it. Using ApplicationName")
        # here makes sense to use get with default value "" ->
        # so that it doesn't raise an exception in the next lines
        # (in case ApplicationName is not found, get returns None)
        runtype = rp.data['RunParameters']["Setup"].get("ApplicationName", "")
```

As `runtype` is only instantiated inside the loop and if both conditions were not satisfied (though it highly unlikely) it would fail as there wont be any `runtype`